src/vs/platform/remote/node/tunnelService.ts

```diff
@@ -120,12 +120,16 @@ class NodeRemoteTunnel extends Disposable implements RemoteTunnel {
 		}
 
 		localSocket.on('end', () => {
-			this._socketsDispose.delete(localSocket.localAddress);
+			if (localSocket.localAddress) {
```

Comment:
This seems like bad typings, I have filed https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/57246 because I see no mention of `localAddress` being `undefined` in the official docs.